test/models/user_test.rb

```diff
@@ -60,7 +60,7 @@ def setup
   should allow_value(*RFauxFactory.gen_strings(1..50, exclude: [:html, :punctuation, :cyrillic, :utf8]).values).for(:lastname)
   should allow_value('A$+#APRocky').for(:login)
   should allow_value(*valid_name_list).for(:description)
-  should allow_value(*RFauxFactory.gen_strings(1..50, exclude: [:html])).for(:password)
+  should allow_value(*RFauxFactory.gen_strings(1..50, exclude: [:html]).values).for(:password)
```

Comment:
While I was testing this patch, OpenSSL library was erroring out because this code actually tries to store the following array into password:

```
UserTest#test_: User should allow :password to be ‹[:alpha, "NgdMvTEoLiXOWjwlyNYACICCAmahfTJQJHQuYSEPIFSmtWVmP"]›, ‹[:alphanumeric, "sLntjP7kT48Hn7bZ8UPZ5mFQRn8L1GnsHbuY3"]›, ‹[:cjk, "鄢堆阸牬黋垹菍漎凇詹攙啝將貎肏"]›, ‹[:cyrillic, "ӝѧИѷ҅ЁКЙЬЦЦӦЕӊҺэЖ҇аѣҙѣТр҂ѲгӼКЅҧЩѦдҜӋөѨ"]›, ‹[:latin1, "ÉÇþÃíúÕüÑõÖÇëÅîÉøôáòÛèôËÈöò"]›, ‹[:numeric, "150737650672785103224408701503244304007721688444"]›, ‹[:utf8, "𦟇𡫹𨋚𥧎𦊀繐꾆"]›, or ‹[:punctuation, "(}"]›. :
```

I think all these RFauxFactory tests in this file, and in the whole codebase, are incorrect:

```
test/models/hostgroup_test.rb
12:    *RFauxFactory.gen_strings(1..245, exclude: [:html, :punctuation]).values,

test/models/host_test.rb
8:    *RFauxFactory.gen_strings(1..255, exclude: [:html, :utf8]).values,

test/models/organization_test.rb
15:      *RFauxFactory.gen_strings(1..242, exclude: [:html]).values,

test/models/location_test.rb
11:      *RFauxFactory.gen_strings(1..246, exclude: [:html]).values,

test/models/user_test.rb
58:  should allow_value(*RFauxFactory.gen_strings(1..50, exclude: [:html, :punctuation, :cyrillic, :utf8]).values).for(:firstname)
60:  should allow_value(*RFauxFactory.gen_strings(1..50, exclude: [:html, :punctuation, :cyrillic, :utf8]).values).for(:lastname)
63:  should allow_value(*RFauxFactory.gen_strings(1..50, exclude: [:html]).values).for(:password)
65:  should_not allow_value(*RFauxFactory.gen_strings(51)).for(:firstname)
67:  should_not allow_value(*RFauxFactory.gen_strings(51)).for(:lastname)

test/test_helper.rb
50:    *RFauxFactory.gen_strings(1..255, exclude: [:html]).values.map { |x| x.truncate_bytes(255, omission: '') },
```

Shall I just fix these or file a different PR? This line must be changed tho, OpenSSL will fail when array is attempted to be encrypted instead of string.

Reply:
I'd be ok with fixing this in here